dex/runner.go

```diff
@@ -107,13 +112,31 @@ func NewConnectionMaster(c Connector) *ConnectionMaster {
 // not to be "on". If the ConnectionMaster is discarded on error, it is not
 // important which method is used.
 func (c *ConnectionMaster) Connect(ctx context.Context) (err error) {
-	c.init(ctx)
-	c.mtx.Lock()
-	c.wg, err = c.connector.Connect(c.ctx)
-	c.mtx.Unlock()
-	// NOTE: Even if err is non-nil, we can't cancel the internal context
-	// because the connector may be attempting to reconnect. The caller should
-	// decide to let it go or to call Disconnect.
+	if c.On() { // probably a bug in the consumer
+		return errors.New("already running")
+	}
+
+	// Attempt to start the Connector.
+	ctx, cancel := context.WithCancel(ctx)
+	wg, err := c.connector.Connect(ctx)
+	if wg == nil {
+		cancel() // no context leak
+		return err
+	}
```

Comment:
Should this have a custom error? Should just be checking `err != nil`? Does it need two checks?

Reply:
I'm thinking the Connector is buggy if both wg and err are nil.

Reply:
To catch the bug a custom error would be better? i.e. "connection master connect error: %w, err"